osu.Game.Tests/Visual/TestCaseBackgroundScreenBeatmap.cs

```diff
@@ -0,0 +1,421 @@
+            typeof(Player),
+            typeof(UserDimContainer),
+            typeof(OsuScreen)
+        };
+
+        private DummySongSelect songSelect;
+        private DimAccessiblePlayerLoader playerLoader;
+        private DimAccessiblePlayer player;
+        private DatabaseContextFactory factory;
+        private BeatmapManager manager;
+        private RulesetStore rulesets;
+
+        private ScreenStackCacheContainer screenStackContainer;
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost host)
+        {
+            factory = new DatabaseContextFactory(LocalStorage);
+            factory.ResetDatabase();
+
+            using (var usage = factory.Get())
+                usage.Migrate();
+
+            factory.ResetDatabase();
+
+            using (var usage = factory.Get())
+                usage.Migrate();
+
+            Dependencies.Cache(rulesets = new RulesetStore(factory));
+            Dependencies.Cache(manager = new BeatmapManager(LocalStorage, factory, rulesets, null, null, host, Beatmap.Default));
+            Dependencies.Cache(new OsuConfigManager(LocalStorage));
+
+            Beatmap.SetDefault();
+        }
+
+        [SetUp]
+        public virtual void SetUp()
+        {
+            Schedule(() =>
+            {
+                manager.Delete(manager.GetAllUsableBeatmapSets());
+                var temp = TestResources.GetTestBeatmapForImport();
+                manager.Import(temp);
+            });
+        }
+
+        /// <summary>
+        /// Check if PlayerLoader properly triggers background dim previews when a user hovers over the visual settings panel.
+        /// </summary>
+        [Test]
+        public void PlayerLoaderSettingsHoverTest()
+        {
+            createSongSelect();
+            AddStep("Start player loader", () => songSelect.Push(playerLoader = new DimAccessiblePlayerLoader(player = new DimAccessiblePlayer())));
+            AddUntilStep(() => playerLoader?.IsLoaded ?? false, "Wait for Player Loader to load");
+            AddAssert("Background retained from song select", () => songSelect.IsBackgroundCurrent());
+            AddStep("Trigger background preview", () =>
+            {
+                InputManager.MoveMouseTo(playerLoader.ScreenPos);
+                InputManager.MoveMouseTo(playerLoader.VisualSettingsPos);
+            });
+
+            waitForDim();
+            AddAssert("Screen is dimmed", () => songSelect.IsBackgroundDimmed());
+        }
+
+        /// <summary>
+        /// In the case of a user triggering the dim preview the instant player gets loaded, then moving the cursor off of the visual settings:
+        /// The OnHover of PlayerLoader will trigger, which could potentially trigger an undim unless checked for in PlayerLoader.
+        /// We need to check that in this scenario, the dim is still properly applied after entering player.
+        /// </summary>
+        [Test]
+        public void PlayerLoaderTransitionTest()
```

Comment:
have you checked this is testing what you say it is? for me the timings don't seem to be correct.